montepy/mcnp_object.py

```diff
@@ -100,22 +101,34 @@ class MCNP_Object(ABC, metaclass=_ExceptionContextAdder):
         For init removed ``comments``, and added ``parser`` as arguments.
 
     :param input: The Input syntax object this will wrap and parse.
-    :type input: Input
+    :type input: Union[Input, str]
     :param parser: The parser object to parse the input with.
-    :type parser: MCNP_Lexer
+    :type parser: MCNP_Parser
+    """
+
+    """
+    The block type this input comes from.
     """
 
     def __init__(
         self,
-        input: montepy.input_parser.mcnp_input.Input,
+        input: Union[montepy.input_parser.mcnp_input.Input, str],
         parser: montepy.input_parser.parser_base.MCNP_Parser,
     ):
+        try:
+            self._BLOCK_TYPE
+        except AttributeError:
+            self._BLOCK_TYPE = montepy.input_parser.block_type.BlockType.DATA
         self._problem_ref = None
         self._parameters = ParametersNode()
         self._input = None
         if input:
-            if not isinstance(input, montepy.input_parser.mcnp_input.Input):
+            if not isinstance(input, (montepy.input_parser.mcnp_input.Input, str)):
                 raise TypeError("input must be an Input")
```

Comment:
" or str"

Reply:
Sometimes I silently ignore optional types in typeErrors to discourage certain behavior, but I think this behavior should be encouraged.